Extension/src/LanguageServer/ui_new.ts

```diff
@@ -0,0 +1,551 @@
+
+        // TODO: Integrate with Tarik's feature to determine if compiler/bare-intellisense is configured
+        if (settings.intelliSenseEngine === "disabled") {
+            this.intelliSenseStatusBarItem.text = this.missingIntelliSenseText;
+            this.intelliSenseStatusBarItem.severity = vscode.LanguageStatusSeverity.Warning;
+            this.intelliSenseStatusBarItem.command = {
+                command: "C_Cpp.CheckForCompiler",
+                title: localize("intellisense.select.text", "Select a Compiler")
+            };
+            return;
+        }
+
+        this.intelliSenseStatusBarItem.busy = val;
+
+        if (this.flameTimeout) {
+            clearTimeout(this.flameTimeout);
+        }
+
+        if (val) {
+            this.intelliSenseStatusBarItem.text = "$(flame)";
+            this.intelliSenseStatusBarItem.detail = this.updatingIntelliSenseText;
+            this.intelliSenseStatusBarItem.severity = vscode.LanguageStatusSeverity.Warning;
+        } else {
+            this.flameTimeout = setTimeout(() => {
+                if (this.intelliSenseStatusBarItem) {
+                    this.intelliSenseStatusBarItem.text = this.idleIntelliSenseText;
+                    this.intelliSenseStatusBarItem.detail = "";
+                    this.intelliSenseStatusBarItem.severity = vscode.LanguageStatusSeverity.Warning;
+                }
+            }, this.iconDelayTime);
+        }
+        this.intelliSenseStatusBarItem.command = {
+            command: "C_Cpp.RestartIntelliSenseForFile",
+            title: localize("rescan.intellisense.text", "Rescan"),
+            tooltip: localize("rescan.intellisense.tooltip", "Rescan IntelliSense")
+        };
+
+    }
+
+    private codeAnalysisCurrentState(): string {
+        const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
+        const state: string = settings.codeAnalysisRunAutomatically && settings.clangTidyEnabled ? "Automatic" : "Manual";
+        return state;
+    }
+
+    private setIsRunningCodeAnalysis(val: boolean): void {
+        if (this.isRunningCodeAnalysis && !val) {
+            this.codeAnalysisTotal = 0;
+            this.codeAnalysisProcessed = 0;
+        }
+        this.isRunningCodeAnalysis = val;
+        this.codeAnalysisStatusBarItem.busy = val;
+        this.codeAnalysisStatusBarItem.text = val ? "Code Analysis: Running" : `Code Analysis State: ${this.codeAnalysisCurrentState()}`;
+        this.codeAnalysisStatusBarItem.command = val ? {
+            command: "C_Cpp.ShowActiveCodeAnalysisCommands",
+            title: localize("c.cpp.codeanalysis.statusbar.showCodeAnalysisOptions", "Options")
+        } : {
+            command: "C_Cpp.ShowIdleCodeAnalysisCommands",
```

Comment:
As mentioned elsewhere, I'm concerned that hiding the "idle" code analysis commands when it's active is potentially confusing because the commands are still valid/usable while active.

Reply:
I've addressed it here the way I proposed
https://github.com/microsoft/vscode-cpptools/blob/42e26923401ff5ee5e73d678f709ae395613390d/Extension/src/LanguageServer/ui_new.ts#L483
https://github.com/microsoft/vscode-cpptools/blob/42e26923401ff5ee5e73d678f709ae395613390d/Extension/src/LanguageServer/client.ts#L2766